src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1668,26 +1668,53 @@ private DiagnosticInfo GetBadEventUsageDiagnosticInfo(EventSymbol eventSymbol)
                 new CSDiagnosticInfo(ErrorCode.ERR_BadEventUsageNoField, leastOverridden);
         }
 
-        internal static bool AccessingAutoPropertyFromConstructor(BoundPropertyAccess propertyAccess, Symbol fromMember)
+        // PROTOTYPE(semi-auto-props): Re-evaluate if this helper is needed. There is one callsite only.
+        internal static bool IsEquivalentToBackingFieldRead(BoundPropertyAccess propertyAccess, Symbol fromMember)
         {
-            return AccessingAutoPropertyFromConstructor(propertyAccess.ReceiverOpt, propertyAccess.PropertySymbol, fromMember);
+            var propertySymbol = propertyAccess.PropertySymbol;
+            var receiver = propertyAccess.ReceiverOpt;
+            if (!propertySymbol.IsDefinition && propertySymbol.ContainingType.Equals(propertySymbol.ContainingType.OriginalDefinition, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
+            {
+                propertySymbol = propertySymbol.OriginalDefinition;
+            }
+
+            return propertySymbol is SourcePropertySymbolBase sourceProperty &&
+                sourceProperty.GetMethod is SourcePropertyAccessorSymbol { IsEquivalentToBackingFieldAccess: true } &&
+                TypeSymbol.Equals(sourceProperty.ContainingType, fromMember.ContainingType, TypeCompareKind.ConsiderEverything2) &&
```

Comment:
While we're here, should we update `ConsiderEverything2`? I think it should probably be `ConsiderEverything`, no 2.

Reply:
They should be the same but I was keeping the original code. Will change if @AlekseyTs agrees

Reply:
> They should be the same but I was keeping the original code. Will change if @AlekseyTs agrees

Yeah, ConsiderEverything2 is an artifact from a refactor we did a couple of years ago to specify everywhere, and we generally try to remove it/not use it for new code.

Reply:
```suggestion
                TypeSymbol.Equals(sourceProperty.ContainingType, fromMember.ContainingType, TypeCompareKind.ConsiderEverything) &&
```

Reply:
If I was working on the change, I would leave this part as is. Even though we are making changes close by, we haven't really done any work to figure out what comparison kind is supposed to be used here. I actually think that ignoring everything might be more appropriate (see the ```if``` above that deals with nullability differences, for example). Since we haven't done the work, let's not remove the tag.

Reply:
> ConsiderEverything2 is an artifact from a refactor we did a couple of years ago to specify everywhere, and we generally try to remove it/not use it for new code.

The goal of using ```ConsiderEverything2 ``` is to come back and reevaluate what value should actually be used, not just replace it with ```ConsiderEverything``` when we are making changes close by.

Reply:
> > ConsiderEverything2 is an artifact from a refactor we did a couple of years ago to specify everywhere, and we generally try to remove it/not use it for new code.
> 
> The goal of using `ConsiderEverything2 ` is to come back and reevaluate what value should actually be used, not just replace it with `ConsiderEverything` when we are making changes close by.

Yes, I'm aware. And my thought was, since we were touching this, it might be a good opportunity to do that evaluation. If you don't think we should do that now, though, I'm ok with it.